openblas build fails on archlinux-latest-minimal, fedora-38-minimal

<div id="comment:0"></div>

https://github.com/mkoeppe/sage/actions/runs/3853109922/jobs/6583067996

```
2023-01-07T17:31:18.8289483Z gcc -g -O2 -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=2 -DMAX_PARALLEL_NUMBER=1 -DUSE_TLS -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.21\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I. -O2 -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=2 -DMAX_PARALLEL_NUMBER=1 -DUSE_TLS -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.21\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I.. -Wl,-rpath-link,/sage/local/lib -L/sage/local/lib -Wl,-rpath,/sage/local/lib -Wl,-rpath-link,/sage/local/lib -L/sage/local/lib -Wl,-rpath,/sage/local/lib  -w -o linktest linktest.c ../libopenblasp-r0.3.21.so -L/sage/local/lib/../lib -L/sage/local/lib/../lib -L/sage/local/lib/gcc/x86_64-pc-linux-gnu/12.2.0 -L/sage/local/lib/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../lib -L/lib/../lib -L/usr/lib/../lib -L/sage/local/lib -L/sage/local/lib -L/sage/local/lib/gcc/x86_64-pc-linux-gnu/12.2.0/../../..  -lgfortran -lm -lquadmath -lm -lc  && echo OK.
2023-01-07T17:31:18.8289656Z /usr/sbin/ld: /tmp/cc0ihrCo.o: in function `main':
2023-01-07T17:31:18.8289973Z /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:191: undefined reference to `iparam2stage_'
2023-01-07T17:31:18.8290308Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:192: undefined reference to `ilaenv2stage_'
2023-01-07T17:31:18.8290629Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:204: undefined reference to `spotri_'
2023-01-07T17:31:18.8290927Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:216: undefined reference to `dpotri_'
2023-01-07T17:31:18.8291237Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:228: undefined reference to `cpotri_'
2023-01-07T17:31:18.8291543Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:240: undefined reference to `zpotri_'
2023-01-07T17:31:18.8291843Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:241: undefined reference to `ilaenv_'
2023-01-07T17:31:18.8292197Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:242: undefined reference to `ieeeck_'
2023-01-07T17:31:18.8292494Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:243: undefined reference to `lsamen_'
2023-01-07T17:31:18.8292792Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:244: undefined reference to `iparmq_'
2023-01-07T17:31:18.8293104Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:245: undefined reference to `ilaprec_'
2023-01-07T17:31:18.8293414Z /usr/sbin/ld: /sage/local/var/tmp/sage/build/openblas-0.3.21/src/exports/linktest.c:246: undefined reference to `ilatrans_'
...
```



CC:  @antonio-rojas

Component: **packages: standard**

_Issue created by migration from https://trac.sagemath.org/ticket/34899_

--- a/build/pkgs/openblas/spkg-install.in
+++ b/build/pkgs/openblas/spkg-install.in
@@ -34,8 +34,13 @@ fi
 
 echo "Building OpenBLAS: $MAKE $OPENBLAS_CONFIGURE"
 
-# Do not emit "-j" options
-OPENBLAS_CONFIGURE+=" MAKE_NB_JOBS=0"
+if $MAKE --version | grep -q -F '3.81'; then
+    # Work around https://savannah.gnu.org/bugs/?15919
+    OPENBLAS_CONFIGURE+=" MAKE_NB_JOBS=1"
+else
+    # Do not emit "-j" options; use jobserver
+    OPENBLAS_CONFIGURE+=" MAKE_NB_JOBS=0"
+fi
 
 # Ensure USE_TLS=1 ; see https://github.com/sagemath/sage/issues/27256
 OPENBLAS_CONFIGURE="$OPENBLAS_CONFIGURE USE_TLS=1"

--- a/build/pkgs/openblas/spkg-install.in
+++ b/build/pkgs/openblas/spkg-install.in
@@ -35,7 +35,7 @@ fi
 echo "Building OpenBLAS: $MAKE $OPENBLAS_CONFIGURE"
 
 # Do not emit "-j" options
-export MAKE_NB_JOBS=0
+OPENBLAS_CONFIGURE+=" MAKE_NB_JOBS=0"
 
 # Ensure USE_TLS=1 ; see https://github.com/sagemath/sage/issues/27256
 OPENBLAS_CONFIGURE="$OPENBLAS_CONFIGURE USE_TLS=1"

--- a/build/pkgs/openblas/spkg-install.in
+++ b/build/pkgs/openblas/spkg-install.in
@@ -34,6 +34,9 @@ fi
 
 echo "Building OpenBLAS: $MAKE $OPENBLAS_CONFIGURE"
 
+# Do not emit "-j" options
+export MAKE_NB_JOBS=0
+
 # Ensure USE_TLS=1 ; see https://github.com/sagemath/sage/issues/27256
 OPENBLAS_CONFIGURE="$OPENBLAS_CONFIGURE USE_TLS=1"
 

--- a/build/pkgs/openblas/spkg-install.in
+++ b/build/pkgs/openblas/spkg-install.in
@@ -45,7 +45,7 @@ fi
 # Ensure USE_TLS=1 ; see https://github.com/sagemath/sage/issues/27256
 OPENBLAS_CONFIGURE="$OPENBLAS_CONFIGURE USE_TLS=1"
 
-if ! (sdh_make libs netlib shared $OPENBLAS_CONFIGURE); then
+if ! (sdh_make libs $OPENBLAS_CONFIGURE && sdh_make netlib $OPENBLAS_CONFIGURE && sdh_make shared $OPENBLAS_CONFIGURE); then
     if [[ $OPENBLAS_CONFIGURE == *"TARGET"* ]]; then
         sdh_die "Error building OpenBLAS"
     else
@@ -55,7 +55,7 @@ if ! (sdh_make libs netlib shared $OPENBLAS_CONFIGURE); then
         echo "Error building OpenBLAS"
         echo "Retrying building OpenBLAS: $MAKE $OPENBLAS_CONFIGURE"
         sdh_make clean
-        sdh_make libs netlib shared $OPENBLAS_CONFIGURE
+        sdh_make libs $OPENBLAS_CONFIGURE && sdh_make netlib $OPENBLAS_CONFIGURE && sdh_make shared $OPENBLAS_CONFIGURE
     fi
 fi
 

